packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1590,6 +1592,42 @@ function shouldRemainOnFallback(
   );
 }
 
+function getRemainingWorkInPrimaryTree(
+  workInProgress,
+  currentChildExpirationTime,
+  renderExpirationTime,
+) {
+  if (currentChildExpirationTime < renderExpirationTime) {
+    // The highest priority remaining work is not part of this render. So the
+    // remaining work has not changed.
+    return currentChildExpirationTime;
+  }
+  if ((workInProgress.mode & BlockingMode) !== NoMode) {
+    // The highest priority remaining work is part of this render. Since we only
+    // keep track of the highest level, we don't know if there's a lower
+    // priority level scheduled. As a compromise, we'll render at a really low
+    // priority that includes all the updates.
+    // TODO: If expirationTime were a bitmask where each bit represents a
+    // separate task thread, this would be: currentChildBits & ~renderBits
+    // TODO: We used to track the lowest pending level for the whole root, but
+    // we removed it to simplify the implementation. It might be worth adding
+    // it back for this use case, to avoid redundant passes.
```

Comment:
@sebmarkbage I think I'm going to do this

Reply:
Instead of tracking it globally, why won't we just track it on this suspense boundary in the state?

Reply:
I'm going to track the "suspended time" locally but to track the "last pending time" locally I'd have to add stuff to the `setState` path. Doesn't seem worth it since the bitmask version will be much simpler.